add scikit-posthocs recipe

huidongchen opened the pull request:
<!-- 
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with 
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`

If your PR doesn't fall into those catagories please contact 
the full review team `@conda-forge/staged-recipes`.
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/scikit-posthocs```) and found some lint.

Here's what I've got...


For **recipes/scikit-posthocs**:

 * The recipe could do with some maintainers listed in the `extra/recipe-maintainers` section.
 * Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [10]
 * The recipe `license` should not include the word "License".

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/scikit-posthocs```) and found some lint.

Here's what I've got...


For **recipes/scikit-posthocs**:

 * Failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint .` from the recipe directory.

beckermr commented:
Try adding pip to the host env.

huidongchen commented:
> Try adding pip to the host env.

Hi Matthew, thanks so much for your quick reply!

I added 'pip' as you suggested. But it failed again with a different error. It looks like it has something to do with the 'setup.py' file (I'm not the developer. But I'm trying to write a conda package recipe dependent on 'scikit-posthocs') . 

So I forked their repo and modified the original setup file. Trying it again now. Fingers crossed.

huidongchen commented:
> > Try adding pip to the host env.
> 
> Hi Matthew, thanks so much for your quick reply!
> 
> I added 'pip' as you suggested. But it failed again with a different error. It looks like it has something to do with the 'setup.py' file (I'm not the developer. But I'm trying to write a conda package recipe dependent on 'scikit-posthocs') .
> 
> So I forked their repo and modified the original setup file. Trying it again now. Fingers crossed.

@beckermr Unfortunately it failed again.. Could you enlighten me re the new error? Thanks!

beckermr commented:
Ok. So the recipe cannot fine the license file. You will need to package a copy of the upstream license with the recipe.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/scikit-posthocs```) and found some lint.

Here's what I've got...


For **recipes/scikit-posthocs**:

 * Failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint .` from the recipe directory.

huidongchen commented:
The recipe passed the local test "conda smithy recipe-lint ." . I don't understand why it still failed in the step 'conda-forge-linter'

beckermr commented:
You probably have to wait for the linter to run again.

beckermr commented:
We can do this!

@conda-forge-admin, please lint

huidongchen commented:
> We can do this!
> 
> @conda-forge-admin, please lint

Many thanks for your great help! Happy new year!

beckermr commented:
You too! Let’s hope the bot is running. This delay has been pretty long.

beckermr commented on recipes/scikit-posthocs/meta.yaml:
```diff
@@ -0,0 +1,57 @@
+
+extra:
+  recipe-maintainers:
+    - Huidong Chen
```
This should be your GitHub username.

beckermr commented on recipes/scikit-posthocs/meta.yaml:
```diff
@@ -0,0 +1,57 @@
+# Jinja variables help maintain the recipe as you'll update the version only here.
+# Using the name variable with the URL in line 13 is conviniet
+# when copying and pasting from another recipe, but not really needed.
+
```
All of the text above can be removed.

beckermr commented on recipes/scikit-posthocs/meta.yaml:
```diff
@@ -0,0 +1,57 @@
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - python
```
Python should only be in the host environment.

beckermr commented on recipes/scikit-posthocs/meta.yaml:
```diff
@@ -0,0 +1,57 @@
+  run:
+    - python
+    - scipy
+    - numpy
```
This line needs to be a jinja2 pin compatible expression. Google for the conda forge docs and the proper thing is listed there.

beckermr commented on recipes/scikit-posthocs/meta.yaml:
```diff
@@ -0,0 +1,57 @@
+    - seaborn
+
+test:
+  # Python imports
```
This comment can be removed too.

beckermr commented on recipes/scikit-posthocs/meta.yaml:
```diff
@@ -0,0 +1,57 @@
+  version: "0.4.0"
+
+source:
+  url: https://github.com/huidongchen/scikit-posthocs/archive/master.zip
```
Wait is this pulling from your fork? We cannot do that. We need to move any patch to master or store it and apply it here.

huidongchen replied:
Yeah, the original setup file caused some errors during the building. So I had to modify that part to make it work. We are currently kind of rushing to finish a bioconda package which is dependent on scikit-posthocs. Should I contact the package 'scikit-posthocs' developer and wait for them to update it? Or are there any other faster solutions you would suggest?

beckermr replied:
We add a file to repo that patches the source before installation. See the conda build documentation. 

This would be a file with the text produced by git diff between your commit and the upstream repo.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/scikit-posthocs```) and found some lint.

Here's what I've got...


For **recipes/scikit-posthocs**:

 * Failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint .` from the recipe directory.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/scikit-posthocs```) and found some lint.

Here's what I've got...


For **recipes/scikit-posthocs**:

 * Failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint .` from the recipe directory.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/scikit-posthocs```) and found it was in an excellent condition.

huidongchen commented:
Hi @beckermr , I resolved the issues you pointed out earlier. But it still failed. It seems to have something to do with the original setup file. Could you let me know how I should proceed here? Thanks!

beckermr commented:
So what you need to do is the make a file via diffing the original setup.py and your changes. 

Something like

```
diff -u <old> <new>
```

should do the trick though you may have to adjust the patching paths by hand. 

Then you can add this patch file to the recipe according to the docs here: https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#patches
